ProjFS.Mac/PrjFSKextTests/MockProc.cpp

```diff
@@ -1,23 +1,24 @@
 #include "MockProc.hpp"
-#include <string.h>
+#include <string>
+
+static std::string s_procName;
 
 int proc_pid(proc_t)
 {
     return 1;
 }
 
-char* return_proc_name;
-
 void proc_name(int pid, char * buf, int size)
 {
-    if (return_proc_name != nullptr)
-    {
-        strcpy(buf, return_proc_name);
-    }
+    strlcpy(buf, s_procName.c_str(), MIN(size, MAXPATHLEN));
```

Comment:
Checking for `MAXPATHLEN` isn't really appropriate here. There is a `MAXCOMLEN` limitation in the kernel, but for our purposes I think we can ignore that for the sake of simplicity:

```
strlcpy(buf, s_procName.c_str(), size);
```

Reply:
Sigh yes :-), copy/paste error